src/threads.cpp

```diff
@@ -97,7 +97,6 @@ void * reading_thread(void *arg) {
 				/* insert readings into channel queues */
 				if (n>0)
 				for (MeterMap::iterator ch = mapping->begin(); ch!=mapping->end(); ch++) {
-					Reading *add = NULL;
```

Comment:
fine. if the below mentioned if (add==...) is the only usage here. then remove it. +1